dace/frontend/python/newast.py

```diff
@@ -3330,12 +3330,18 @@ def _visit_assign(self, node, node_target, op, dtype=None, is_return=False):
                     true_name = defined_vars[name]
                     while len(tokens) > 1:
                         true_name = true_name + '.' + tokens.pop(0)
-                        if true_name not in self.sdfg.arrays:
+                        if true_name not in defined_arrays:
                             break
                     if tokens:  # The non-struct remainder will be considered an attribute
                         attribute_name = '.'.join(tokens)
-                        raise DaceSyntaxError(
-                            self, target, f'Cannot assign to attribute "{attribute_name}" of variable "{true_name}"')
+                        combined_name = true_name + '.' + attribute_name
+                        if (combined_name in defined_arrays and true_name in defined_arrays
+                                and isinstance(defined_arrays[true_name], data.Structure)):
+                            true_name = combined_name
+                        else:
```

Comment:
but now this part is not necessary

Reply:
The syntax error warning or the entire check?

Reply:
The entire check can be reverted to the original code because the while loop covers the same condition.

Reply:
Changed